packages/zendesk-adapter/src/change_validators/one_translation_per_locale.ts

```diff
@@ -0,0 +1,106 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import _ from 'lodash'
+import {
+  ChangeValidator,
+  getChangeData, InstanceElement,
+  isAdditionOrModificationChange, isInstanceElement, ReferenceExpression,
+} from '@salto-io/adapter-api'
+import Joi from 'joi'
+import { createSchemeGuardForInstance, getParents } from '@salto-io/adapter-utils'
+// eslint-disable-next-line no-restricted-imports
+import { findDuplicates } from '@salto-io/lowerdash/dist/src/collections/array'
+
+
+const ARTICLE_TRANSLATION_TYPE_NAME = 'article_translation'
+const CATEGORY_TRANSLATION_TYPE_NAME = 'category_translation'
+const SECTION_TRANSLATION_TYPE_NAME = 'section_translation'
+
+type Translation = InstanceElement & {
+    value: {
+        locale: string
+    }
+}
+type TranslationParent = InstanceElement & {
+    value: {
+        translations: ReferenceExpression[]
+    }
+}
+
+const TRANSLATION_SCHEMA = Joi.object({
+  locale: Joi.string().required(),
+}).unknown(true).required()
+
+const TRANSLATION_PARENT_SCHEMA = Joi.object({
+  translations: Joi.array().required(),
+}).unknown(true).required()
+
+const isTranslationParent = createSchemeGuardForInstance<TranslationParent>(
+  TRANSLATION_PARENT_SCHEMA, 'Received an invalid value for translation parent'
+)
+const isTranslation = createSchemeGuardForInstance<Translation>(
+  TRANSLATION_SCHEMA, 'Received an invalid value for translation'
+)
+
+/**
+ * returns true if there are translations with the same locale -> meaning that they are not unique
+ */
+const hasDuplicateTranslations = (parent: InstanceElement): string[] => {
+  if (!isTranslationParent(parent)) {
+    return []
+  }
+  const locales = parent.value.translations
+    .map(referenceExpression => referenceExpression.value)
+    .filter(translation => isTranslation(translation))
+    .map(translation => translation.value.locale)
+  return findDuplicates(locales)
+}
+
+export const oneTranslationPerLocaleValidator: ChangeValidator = async changes => {
+  const relevantInstances = changes
+    .filter(isAdditionOrModificationChange)
+    .map(getChangeData)
+    .filter(isInstanceElement)
+    .filter(instance =>
+      [ARTICLE_TRANSLATION_TYPE_NAME,
+        CATEGORY_TRANSLATION_TYPE_NAME,
+        SECTION_TRANSLATION_TYPE_NAME].includes(instance.elemID.typeName))
+
+  // it is a set to avoid duplicate of parents from translations which have the same parent
+  const parentInstances = relevantInstances
+    .filter(instance => isTranslation(instance))
+    .flatMap(translationInstance => {
+      const parents = getParents(translationInstance)
+      return !_.isEmpty(parents) ? parents : []
+    })
```

Comment:
nit: I don't think you need this check (`getParents` always returns an array)
```suggestion
    .flatMap(getParents)
```